### PR TITLE
S03_US23 : Listed jobs from a company when connected

### PR DESCRIPTION
**Pull request by Erdnaxela45**

### CODE REVIEW

**Review by maxime-guichard — DISMISSED** [review]

File: src/Repository/JobRepository.php [hunk 1]
```diff
@@ -19,6 +20,31 @@ public function __construct(ManagerRegistry $registry)
...
+            )
+            ->from('job', 'j')
+            ->join('job', 'company', 'company.id = job.company_id')
+            ->where('company.id=:id')
+            ->setParameter('id', $id)
+            ->orderBy('job.registeredAt', 'DESC')
+            ->getQuery()
+            ->getResult()
+        ;
+    }
```
Comment (LucyMerlier) [inline src/Repository/JobRepository.php:46]: you don't need to do that to retrieve info for the jobs that a company has posted => $jobs is already a class property of Company, so their information can simply be accessed using `$company->getJobs()`

File: templates/company/list.html.twig [hunk 2]
```diff
@@ -0,0 +1,44 @@
+{% extends 'layout_company.html.twig' %}
+
+{% block title %}{{ parent() }} - Offres{% endblock %}
```
Comment (LucyMerlier) [inline templates/company/list.html.twig:3]: the name of the file is not very clear : "company/list.html.twig" looks like this view is used to list all companies, but it lists all the jobs associated with one company => maybe "company/list_jobs.html.twig" is a bit more eloquent?

File: templates/component/_navbar_user.html.twig [hunk 3]
```diff
@@ -6,7 +6,10 @@
                 <a class="nav-link ms-5 me-5 active" href="{{ path('home') }}">Accueil</a>
             </li>
             <li class="nav-item">
-                <a class="nav-link ms-5 me-5 active" href="#">Offres</a>
+                <a class="nav-link ms-5 me-5 active" href="
+                {% if is_granted('ROLE_COMPANY') %}
+                    {{ path('company_jobs') }}
+                {% endif %}">Offres</a>
```
Comment (LucyMerlier) [inline templates/component/_navbar_user.html.twig:12]: so this link does nothing if we don't have "ROLE_COMPANY"?

**Review by LucyMerlier — CHANGES_REQUESTED** [review]
see comments

File: src/DataFixtures/UserFixtures.php [hunk 4]
```diff
@@ -7,6 +7,7 @@
 use Doctrine\Persistence\ObjectManager;
 use App\Entity\User;
 use Symfony\Component\Security\Core\Encoder\UserPasswordEncoderInterface;
+use Doctrine\Common\DataFixtures\DependentFixtureInterface;
```
Comment (LucyMerlier) [inline src/DataFixtures/UserFixtures.php:10]: you have this "use" twice => delete one of them :wink:

File: src/Entity/User.php [hunk 5]
```diff
@@ -50,6 +50,7 @@ class User implements UserInterface
     private string $phone;
 
     /**
+
```
Comment (LucyMerlier) [inline src/Entity/User.php:53]: why this empty line?

**Review by LucyMerlier — CHANGES_REQUESTED** [review]
see comments

**Pull request closed without merging**